rllib/core/learner/torch/torch_learner.py

```diff
@@ -290,6 +290,9 @@ def build(self) -> None:
         flags, so that `_make_module()` can place the created module on the correct
         device. After running super() it will wrap the module in a TorchDDPRLModule
         if `_distributed` is True.
+        Note, in inherited classes it is advisable to call the parent's `build()`
```

Comment:
Btw, I'm not sure why we don't move the entire code before the `super().build()` call below into the c'tor already. All we do here is to determine the GPU device - if any - based on config settings, which are all available already at c'tor time.

Then we don't need this comment here and we should also add the `@OverrideToImplementCustomLogic_CallToSuperRecommended` decorator on top of this method.

Reply:
I am with you on this. 

However, the comment I added there was b/c when calling the `build()` in inherited classes at the beginning in the inherited `build()` method `configure_optimizers_for_module()` gets called before variables were defined (example: variables defined in the `build()` of the `SACLearner`, then calling `super().build()` before defining the variables in the `SACLearner` would have led to a case where the `Learner.build()` would call all `configure_optimizers_for_module` methods that were overriden by the `SACTorchLearner` and needed variables defined in the `SACLearner` - and these were not yet defined. So first define variables then call `build()` and to keep the knowledge I added it here.)

Reply:
Ok, let's test this (and change only, if possible) before we merge ...

Reply:
That makes sense. So it's not about **whether** to call super, but about **when** to call it.

The call to super must be after(!) all parameters are available (for the optimizers to know).

Reply:
Exactly! Let's keep this in mind for documentation! I am sure users who derive from this class will run into it.